### PR TITLE
[docs] Replace @mui/styles in x-data-grid-generator

### PR DESCRIPTION
**Pull request by m4theushw**
Part of #2991

After this PR there will only a few usages of `@mui/styles` remaining.

Preview: https://deploy-preview-3560--material-ui-x.netlify.app/components/data-grid/demo/

Preview: https://deploy-preview-3560--material-ui-x.netlify.app/storybook/?path=/story/datagridpro-test-rows--edit-rows-control

### CODE REVIEW

File: packages/grid/x-data-grid-generator/src/renderer/renderTotalPrice.tsx [hunk 1]
```diff
@@ -1,27 +1,24 @@
...
-      },
-      bad: {
-        backgroundColor: alpha(theme.palette.error.main, 0.3),
-      },
-    }),
-  { defaultTheme },
-);
+const Value = styled('div')(({ theme }) => ({
+  width: '100%',
+  height: '100%',
```
Comment (m4theushw) [inline packages/grid/x-data-grid-generator/src/renderer/renderTotalPrice.tsx:8]: The added `height` is to fix a bug created by #3446. It was also added in edit components.

image

image

File: packages/grid/x-data-grid-generator/src/renderer/renderEditCurrency.tsx [hunk 2]
```diff
@@ -1,46 +1,24 @@
...
+  [`& .${autocompleteClasses.inputRoot}`]: {
+    ...theme.typography.body2,
+    padding: '1px 0',
+    height: '100%',
+    '& input': {
+      padding: '0 16px',
+      height: '100%',
+    },
+  },
+}));
```
Comment (alexfauquette) [inline packages/grid/x-data-grid-generator/src/renderer/renderEditCurrency.tsx:19]: Same here

on the preview
image

on the doc
image

File: packages/grid/x-data-grid-generator/src/renderer/renderEditStatus.tsx [hunk 3]
```diff
@@ -89,10 +74,10 @@ function EditStatus(props: GridRenderEditCellParams) {
 
         return (
           <MenuItem key={option} value={option}>
-            <ListItemIcon className={classes.optionIcon}>
+            <ListItemIcon sx={{ minWidth: 40 }}>
```
Comment (alexfauquette) [inline packages/grid/x-data-grid-generator/src/renderer/renderEditStatus.tsx:77]:
```suggestion
            <ListItemIcon sx={{ minWidth: 36 }}>
```

Comment (alexfauquette) [reply packages/grid/x-data-grid-generator/src/renderer/renderEditStatus.tsx:77]: depends if there is a reason to change the previous value

**Review by alexfauquette — COMMENTED** [review]
I did not read all of them. I will finish the reading tomorow

**Comment by alexfauquette** [comment]
Just finished reading. The other files look good to me :)

File: packages/grid/x-data-grid-generator/src/renderer/renderEditContractType.tsx [hunk 4]
```diff
@@ -3,31 +3,9 @@ import { GridRenderCellParams, GridEvents } from '@mui/x-data-grid';
...
-        minWidth: 36,
-      },
-      optionText: {
-        overflow: 'hidden',
-      },
-    }),
-  { defaultTheme },
-);
-
 function EditContractType(props: GridRenderCellParams) {
```
[the remark this answers is not in the document]
Comment (m4theushw) [reply packages/grid/x-data-grid-generator/src/renderer/renderEditContractType.tsx]: I removed it in c5dfc1c.

File: packages/grid/x-data-grid-generator/src/renderer/renderEditCountry.tsx [hunk 5]
```diff
@@ -1,46 +1,24 @@
...
+  [`& .${autocompleteClasses.inputRoot}`]: {
+    ...theme.typography.body2,
+    padding: '1px 0',
+    height: '100%',
+    '& input': {
+      padding: '0 16px',
+      height: '100%',
+    },
+  },
+}));
```
[the remark this answers is not in the document]
Comment (m4theushw) [reply packages/grid/x-data-grid-generator/src/renderer/renderEditCountry.tsx]: Fixed in 1420d76.

File: packages/grid/x-data-grid-generator/src/renderer/renderEditCurrency.tsx (hunk 2 above)
Comment (m4theushw) [reply packages/grid/x-data-grid-generator/src/renderer/renderEditCurrency.tsx]: Fixed in 1420d76.

File: packages/grid/x-data-grid-generator/src/renderer/renderEditStatus.tsx (hunk 3 above)
Comment (m4theushw) [reply packages/grid/x-data-grid-generator/src/renderer/renderEditStatus.tsx]: I don't remember why I used 40px but I changed to 36px in f0ad99d.

**Review by alexfauquette — APPROVED** [review]

**Pull request merged**